datalad/tests/test_config.py

```diff
@@ -230,9 +230,19 @@ def test_something(path, new_home):
     # very carefully test non-local config
     # so carefully that even in case of bad weather Yarik doesn't find some
     # lame datalad unittest sections in his precious ~/.gitconfig
-    env = get_home_envvars(new_home)
+
+    # Note: An easier way to test this, would be to just set GIT_CONFIG_GLOBAL
+    # to point somewhere else. However, this appears to not be supported by git
+    # for windows 2.26, which is running on our AppVeyor windows builds ATM.
+    # Hence, stick with changed HOME in this test, but be sure to unset a
+    # possible GIT_CONFIG_GLOBAL in addition.
+
+    patched_env = os.environ.copy()
+    patched_env.pop('GIT_CONFIG_GLOBAL', None)
+    patched_env.update(get_home_envvars(new_home))
     with patch.dict('os.environ',
```

Comment:
@adswa 

Yes, if `GIT_CONFIG_GLOBAL` works, this can be changed. However, We'd need to either raise min. git version on all platforms or make that kind of approach conditional on the version we find.

Reply:
> Unless we want to raise git versions everywhere
Personally, I wouldn't mind that. I just checked whether all build images would allow for this. Its possible on all images on Appveyor (Mac, Linux, Windows; Git 2.33) and all virtual environments for GitHub workflows (Git 2.34). The odd one out is Travis, which currently only goes up to 2.29

Reply:
Yeah, it's possible for us and our CI builds. But it has implications for where datalad can be deployed. Serve/Cluster/etc. infrastructure tends to not run on latest versions of our dependencies and this dependency change would be introduced merely for our test setup, not for any datalad feature. So, raising git version on Linux seems wrong to me.

As mostly "consumer" systems, would be fine for Win/OSX, but as far as I am aware we can't specify min versions per platform.

Reply:
> The odd one out is Travis, which currently only goes up to 2.29

Oh, but I believe on travis we can just do ``sudo apt-get install git`` to get the most recent release

Reply:
> this dependency change would be introduced merely for our test setup, not for any datalad feature

yeah, exactly. I don't see a need to raise the minimal version requirements for a datalad deployment, ``GIT_MIN_VERSION`` can stay what it is. It would just be relevant for the tests, and we control our test setup.

Reply:
anyway, this is would be out of scope for this PR. I'm happy with the overall change to only minimally temper with ``HOME`` variables in the test setup.